document-generator-prosecution/src/main/java/uk/gov/companieshouse/document/generator/prosecution/mapping/mappers/ApiToDefendantMapper.java

```diff
@@ -10,15 +12,36 @@
 @RequestScope
 @Mapper(componentModel = "spring")
 public interface ApiToDefendantMapper {
+
+    @BeforeMapping
+    default void beforeMapping(@MappingTarget Defendant defendant, DefendantApi defendantApi) {
+        if(defendantApi.getIsCorporateAppointment()) {
+            defendant.setName(defendantApi.getCompanyOfficerDetailsApi().getCompanyName());
+        } else {
+            StringBuilder name = new StringBuilder();
+            String title = defendantApi.getPersonOfficerDetailsApi().getTitle();
+            String forename = defendantApi.getPersonOfficerDetailsApi().getForename();
+            String middleName = defendantApi.getPersonOfficerDetailsApi().getMiddleName();
+            String surname = defendantApi.getPersonOfficerDetailsApi().getSurname();
+
+            if(title != null && title.trim().length() > 0) {
+                name.append(title).append(" ");
+            }
+            if(forename != null && forename.trim().length() > 0) {
+                name.append(forename).append(" ");
+            }
+            if(middleName != null && middleName.trim().length() > 0){
+                name.append(middleName).append(" ");
+            }
+            if(surname != null && surname.trim().length() > 0) {
+                name.append(surname);
+            }
+            defendant.setName(name.toString());
```

Comment:
If surname is null or empty this will have a blank space at the end. The name.toString() should be trimmed to remove trailing whitespace.

Reply:
Done